scripts/endToEndExtraction/verify-testcases.py

```diff
@@ -0,0 +1,215 @@
+        result = str(
+            "kind='" + self.kind + "' parameter='" + self.parameter + "' input='" + self._input +
+            "' output='" + self._output + "' value='" + self.value + "' result='" + self.result + "'"
+        )
+        return result
+
+
+class TestCase:
+    def __init__(self, name):
+        self.name = name
+        self.metadata = None
+        self.traces = []
+
+    def add_trace(self, kind, parameter):
+        trace = Trace(kind, parameter)
+        self.traces.append(trace)
+        return trace
+
+
+class TraceAnalyser:
+    def __init__(self, file):
+        self.file = file
+        self.tests = {}
+        self.ready = False
+
+    def analyse(self):
+        trace_file = open(self.file, "r")
+        trace = None
+        test_case = None
+        for line in trace_file.readlines():
+            test = re.search(r'Entering test case "(.*)"', line, re.M | re.I)
+            if test:
+                test_name = test.group(1)
+                test_case = TestCase(test_name)
+                self.tests[test_name] = test_case
+
+            metadata = re.search(r'\s*metadata:\s*(.*)$', line, re.M | re.I)
+            if metadata:
+                test_case.metadata = json.loads(metadata.group(1))
+                del test_case.metadata["sources"]
+                del test_case.metadata["compiler"]["version"]
+
+            create = re.search(r'CREATE\s*([a-fA-F0-9]*):', line, re.M | re.I)
+            if create:
+                trace = test_case.add_trace("create", create.group(1))
+
+            call = re.search(r'CALL\s*([a-fA-F0-9]*)\s*->\s*([a-fA-F0-9]*):', line, re.M | re.I)
+            if call:
+                trace = test_case.add_trace("call", call.group(1))  # + "->" + call.group(2))
+
+            if not create and not call:
+                self.parse_parameters(line, trace)
+
+        trace_file.close()
+
+        print(self.file + ":", len(self.tests), "test-cases.")
+
+        self.ready = True
+
+    @staticmethod
+    def parse_parameters(line, trace):
+        input = re.search(r'\s*in:\s*([a-fA-F0-9]*)', line, re.M | re.I)
+        if input:
+            trace.input = input.group(1)
+        output = re.search(r'\s*out:\s*([a-fA-F0-9]*)', line, re.M | re.I)
+        if output:
+            trace.output = output.group(1)
+        result = re.search(r'\s*result:\s*([a-fA-F0-9]*)', line, re.M | re.I)
+        if result:
+            trace.result = result.group(1)
+        gas_used = re.search(r'\s*gas\sused:\s*([a-fA-F0-9]*)', line, re.M | re.I)
+        if gas_used:
+            trace.gas = gas_used.group(1)
+        value = re.search(r'\s*value:\s*([a-fA-F0-9]*)', line, re.M | re.I)
+        if value:
+            trace.value = value.group(1)
+
+    def diff(self, analyser):
+        if not self.ready:
+            self.analyse()
+        if not analyser.ready:
+            analyser.analyse()
+
+        intersection = set(self.tests.keys()) & set(analyser.tests.keys())
```

Comment:
Why do you only test the intersection? If a test was deleted from EndToEndTests but not added to semantic tests, then this would not be detected, would it?

Reply:
Maybe I'll write a small script to verify the deletion.

Reply:
Ah, sorry, that's what the removal script is for!